### ISSUE TITLE
Stack traces are lost when log events are seralized

### ISSUE THREAD

**Issue by itaysabato** [log4js-node/log4js-node#671]
When using a cluster, if I get an error that has own properties, its stack trace is not logged.

Example:
```js
const err = Error();
err["I am"] = "all alone";

logger.error(err);
// produces: "{ 'I am': 'all alone' }"
```
Looking at the code, it's clearly because of [this condition](https://github.com/log4js-node/log4js-node/blob/29368cfcdba29596e4aeca80fe5155c2294a3f9a/lib/log4js.js#L96).

There should be a way to serialize additional props along with the stack and message.

**Comment by itaysabato** [comment]
## Proposed solution
In [serialize](https://github.com/log4js-node/log4js-node/blob/29368cfcdba29596e4aeca80fe5155c2294a3f9a/lib/log4js.js#L96), instead do something like:
```js
if (e && e.stack) {
    e = Object.assign({ message: e.message, stack: e.stack }, e);
}
```

And then in [deserialize](https://github.com/log4js-node/log4js-node/blob/29368cfcdba29596e4aeca80fe5155c2294a3f9a/lib/log4js.js#L119) do something like:
```js
if (e && e.stack) {
  const fakeError = new Error(e);
  Object.keys(e).forEach(key => fakeError[key] = e[key])
  e = fakeError;
}
```

**Comment by nomiddlename** [comment]
Thanks for bringing this up, seems like an easy thing to fix.

**Comment by nomiddlename** [comment]
Released to npm in log4js@2.6.0
